src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -190,17 +191,35 @@ protected XmlFile getConfigFile() {
 
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-        hookSecretConfigs = null; // form binding might omit empty lists
         try {
-            req.bindJSON(this, json);
+            BulkChange bc = new BulkChange(this);
+            try {
+                if (json.has("configs")) {
+                    setConfigs(req.bindJSONToList(GitHubServerConfig.class, json.get("configs")));
+                } else {
+                    setConfigs(Collections.emptyList());
+                }
+                if (json.has("hookSecretConfigs")) {
+                    setHookSecretConfigs(req.bindJSONToList(HookSecretConfig.class, json.get("hookSecretConfigs")));
+                } else {
+                    setHookSecretConfigs(Collections.emptyList());
+                }
+                if (json.optBoolean("isOverrideHookUrl", false) && (json.has("hookUrl"))) {
+                    setHookUrl(json.optString("hookUrl"));
+                } else {
+                    setHookUrl(null);
+                }
+                req.bindJSON(this, json);
+                clearRedundantCaches(configs);
```

Comment:
Other than this method call, the whole `configure` method should not even need to be overridden.

Reply:
The only way I found to simplify the data-binding and just do a `req.bindJson` is to change the signature of  `public URL getHookUrl()` to `public String getHookUrl()`. That could be a breaking change. So not sure if that would be acceptable ?  cc @KostyaSha

Reply:
better ask @jglick i forgot databinding nuances

Reply:
> That could be a breaking change. … So not sure if that would be acceptable ?

https://github.com/search?q=org%3Ajenkinsci%20getHookUrl&type=code does not look like it is used from other plugins.